backend/src/models/database.py

```
@@ -46,6 +46,12 @@ class User(db.Model):
     created_at = db.Column(db.DateTime, default=datetime.utcnow)
     updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
     
+    def set_password(self, password):
+        self.password_hash = generate_password_hash(password)
+
+    def check_password(self, password):
```

Ruff (0.11.9)

Missing return type annotation for public function `check_password`

(ANN201)